compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -643,10 +642,12 @@ impl<'context> Elaborator<'context> {
         if let ImplKind::TraitMethod(_, mut constraint, assumed) = ident.impl_kind {
             constraint.apply_bindings(&bindings);
             if assumed {
-                let trait_impl = TraitImplKind::Assumed {
-                    object_type: constraint.typ,
-                    trait_generics: constraint.trait_generics,
+                let trait_generics = TraitGenerics {
+                    ordered: constraint.trait_generics,
+                    named: constraint.associated_types,
```

Comment:
Would it make sense for `TraitConstraint` to have a `TraitGenerics` field instead of these two? But I don't know if it's intentional that they are separate to have clearer names.

Reply:
Yes, I made the change and it lead me to find somewhere where associated types weren't being checked. Fixing that made type hints no longer required in serialize!